src/python-common/ceph/deployment/hostspec.py

```diff
@@ -28,20 +29,22 @@ def __init__(self,
         #: human readable status
         self.status = status or ''  # type: str
 
+        self.maintenance = maintenance
+
     def to_json(self):
         return {
             'hostname': self.hostname,
             'addr': self.addr,
             'labels': self.labels,
-            'status': self.status,
+            'maintenance': self.maintenance,
```

Comment:
intentionally removing `status` here?

Reply:
yeah, `offline` was written to the status, but kept in-memory only. Adding the `status` to `to_json` made the `offline` flag creep into config-key. No idea if mgr/cephadm ever cleans the offline status, if it was persistently written once. A huge potential to break cephadm by writing it into the config-key store.